homeassistant/components/smartthings/light.py

```diff
@@ -0,0 +1,215 @@
+"""
+Support for lights through the SmartThings cloud API.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/smartthings.light/
+"""
+import asyncio
+
+from homeassistant.components.light import (
+    ATTR_BRIGHTNESS, ATTR_COLOR_TEMP, ATTR_HS_COLOR, ATTR_TRANSITION,
+    SUPPORT_BRIGHTNESS, SUPPORT_COLOR, SUPPORT_COLOR_TEMP, SUPPORT_TRANSITION,
+    Light)
+import homeassistant.util.color as color_util
+
+from . import SmartThingsEntity
+from .const import DATA_BROKERS, DOMAIN
+
+DEPENDENCIES = ['smartthings']
+
+
+async def async_setup_platform(
+        hass, config, async_add_entities, discovery_info=None):
+    """Platform uses config entry setup."""
+    pass
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Add lights for a config entry."""
+    broker = hass.data[DOMAIN][DATA_BROKERS][config_entry.entry_id]
+    async_add_entities(
+        [SmartThingsLight(device) for device in broker.devices.values()
+         if is_light(device)], True)
```

Comment:
`async_update` is used to calculate and cache the color-related attributes, since there's computation involved.  `device` has the initial status at this point.  Alternatively, I could call `async_update` from `init`, but this was more concise.